Warnings on master branch: MPI related

Compiling on CentOS with gcc 8.5.0 compiler:
```c
ompi_datatype_module.c: In function ‘ompi_datatype_finalize’:
ompi_datatype_module.c:696:9: warning: unused variable ‘ret’ [-Wunused-variable]
     int ret = OMPI_SUCCESS;
         ^~~
session_create_errhandler.c:28:19: warning: ‘FUNC_NAME’ defined but not used [-Wunused-const-variable=]
 static const char FUNC_NAME[] = "MPI_Session_create_errhandler";
                   ^~~~~~~~~
instance/instance.c: In function ‘ompi_instance_get_num_psets_complete’:
instance/instance.c:941:19: warning: variable ‘rc’ set but not used [-Wunused-but-set-variable]
     pmix_status_t rc;
                   ^~
instance/instance.c: In function ‘ompi_group_from_pset’:
instance/instance.c:1263:19: warning: comparison of distinct pointer types lacks a cast
     if (group_out == MPI_GROUP_NULL) {
                   ^~
```

Got another when building on Mac:
```c
mca_base_var.c:1546:49: warning: implicit conversion from enumeration type 'mca_base_var_flag_t' to different enumeration type 'mca_base_var_syn_flag_t' [-Wenum-conversion]
        mca_base_var_syn_flag_t flags_derived = flags;
```

Hmmm...and now it fails on Mac too!
```text
duplicate symbol '_ompi_isendrecv_context_t_class' in:
    mpi/c/.libs/libmpi_c.a(libmpi_c_profile_la-isendrecv.o)
    mpi/c/.libs/libmpi_c.a(libmpi_c_noprofile_la-isendrecv.o)
duplicate symbol '_ompi_isendrecv_replace_context_t_class' in:
    mpi/c/.libs/libmpi_c.a(libmpi_c_profile_la-isendrecv_replace.o)
    mpi/c/.libs/libmpi_c.a(libmpi_c_noprofile_la-isendrecv_replace.o)
ld: 2 duplicate symbols for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [libmpi.la] Error 1
make[1]: *** [install-recursive] Error 1
make: *** [install-recursive] Error 1
```
This was configured with external libevent, hwloc, pmix, and prrte. No other configure flags given.

Also got one more warning before the failure:
```c
1 warning generated.
communicator/ft/comm_ft_reliable_bcast.c:164:24: warning: unused variable 'tag' [-Wunused-variable]
    mca_btl_base_tag_t tag = descriptor->tag;
                       ^
```

--- a/ompi/mpi/c/isendrecv.c
+++ b/ompi/mpi/c/isendrecv.c
@@ -3,7 +3,7 @@
  * Copyright (c) 2004-2007 The Trustees of Indiana University and Indiana
  *                         University Research and Technology
  *                         Corporation.  All rights reserved.
- * Copyright (c) 2004-2021 The University of Tennessee and The University
+ * Copyright (c) 2004-2022 The University of Tennessee and The University
  *                         of Tennessee Research Foundation.  All rights
  *                         reserved.
  * Copyright (c) 2004-2008 High Performance Computing Center Stuttgart,
@@ -53,7 +53,11 @@ struct ompi_isendrecv_context_t {
 };
 
 typedef struct ompi_isendrecv_context_t ompi_isendrecv_context_t;
+#if OMPI_BUILD_MPI_PROFILING
 OBJ_CLASS_INSTANCE(ompi_isendrecv_context_t, opal_object_t, NULL, NULL);
+#else
+OBJ_CLASS_DECLARATION(ompi_isendrecv_context_t);
+#endif  /* OMPI_BUILD_MPI_PROFILING */
 
 static int ompi_isendrecv_complete_func (ompi_comm_request_t *request)
 {
@@ -66,7 +70,7 @@ static int ompi_isendrecv_complete_func (ompi_comm_request_t *request)
      *
      * Probably need to bring up in the MPI forum.  
      */
-    
+
     if (MPI_PROC_NULL != context->source) {
         OMPI_COPY_STATUS(&request->super.req_status, 
                          context->subreq[0]->req_status, false);
@@ -129,12 +133,12 @@ int MPI_Isendrecv(const void *sendbuf, int sendcount, MPI_Datatype sendtype,
 
         OMPI_ERRHANDLER_CHECK(rc, comm, rc, FUNC_NAME);
     }
-    
+
     crequest = ompi_comm_request_get ();
     if (NULL == crequest) {
         return OMPI_ERR_OUT_OF_RESOURCE;
     }
-    
+
     context = OBJ_NEW(ompi_isendrecv_context_t);
     if (NULL == context) {
         ompi_comm_request_return (crequest);
@@ -142,8 +146,8 @@ int MPI_Isendrecv(const void *sendbuf, int sendcount, MPI_Datatype sendtype,
     }
 
     crequest->context = &context->super;
-    context->subreq[0] = NULL;
-    context->subreq[1] = NULL;
+    context->subreq[0] = MPI_REQUEST_NULL;
+    context->subreq[1] = MPI_REQUEST_NULL;
     context->source = source;
 
     if (source != MPI_PROC_NULL) { /* post recv */
@@ -158,7 +162,7 @@ int MPI_Isendrecv(const void *sendbuf, int sendcount, MPI_Datatype sendtype,
 
     if (dest != MPI_PROC_NULL) { /* send */
         rc = MCA_PML_CALL(isend(sendbuf, sendcount, sendtype, dest,
-                               sendtag, MCA_PML_BASE_SEND_STANDARD, comm, &context->subreq[nreqs++]));
+                                sendtag, MCA_PML_BASE_SEND_STANDARD, comm, &context->subreq[nreqs++]));
         if (MPI_SUCCESS != rc) {
             OBJ_RELEASE(context);
             ompi_comm_request_return (crequest);

--- a/ompi/mpi/c/isendrecv_replace.c
+++ b/ompi/mpi/c/isendrecv_replace.c
@@ -2,7 +2,7 @@
  * Copyright (c) 2004-2007 The Trustees of Indiana University and Indiana
  *                         University Research and Technology
  *                         Corporation.  All rights reserved.
- * Copyright (c) 2004-2020 The University of Tennessee and The University
+ * Copyright (c) 2004-2022 The University of Tennessee and The University
  *                         of Tennessee Research Foundation.  All rights
  *                         reserved.
  * Copyright (c) 2004-2008 High Performance Computing Center Stuttgart,
@@ -57,6 +57,7 @@ struct ompi_isendrecv_replace_context_t {
 
 typedef struct ompi_isendrecv_replace_context_t ompi_isendrecv_replace_context_t;
 
+#if OMPI_BUILD_MPI_PROFILING
 static void ompi_isendrecv_context_constructor(ompi_isendrecv_replace_context_t *context)
 {
     context->packed_size = 0;
@@ -75,6 +76,9 @@ OBJ_CLASS_INSTANCE(ompi_isendrecv_replace_context_t,
                    opal_object_t, 
                    ompi_isendrecv_context_constructor,
                    ompi_isendrecv_context_destructor);
+#else
+OBJ_CLASS_DECLARATION(ompi_isendrecv_replace_context_t);
+#endif /* OMPI_BUILD_MPI_PROFILING */
 
 static int ompi_isendrecv_replace_complete_func (ompi_comm_request_t *request)
 {
@@ -87,7 +91,7 @@ static int ompi_isendrecv_replace_complete_func (ompi_comm_request_t *request)
      *
      * Probably need to bring up in the MPI forum.  
      */
-    
+
     if (MPI_PROC_NULL != context->source) {
         OMPI_COPY_STATUS(&request->super.req_status,
                          context->subreq[0]->req_status, false);
@@ -165,7 +169,7 @@ int MPI_Isendrecv_replace(void * buf, int count, MPI_Datatype datatype,
     if (NULL == crequest) {
         return OMPI_ERR_OUT_OF_RESOURCE;
     }
-    
+
     context = OBJ_NEW(ompi_isendrecv_replace_context_t);
     if (NULL == context) {
         ompi_comm_request_return (crequest);
@@ -205,7 +209,7 @@ int MPI_Isendrecv_replace(void * buf, int count, MPI_Datatype datatype,
         rc = MPI_ERR_UNKNOWN;
         OMPI_ERRHANDLER_RETURN(rc, comm, rc, FUNC_NAME);
     }
-    
+
     if (source != MPI_PROC_NULL) { /* post recv */
         rc = MCA_PML_CALL(irecv(buf, count, datatype,
                                 source, recvtag, comm, &context->subreq[nreqs++]));

